qiskit/transpiler/coupling.py

```diff
@@ -148,14 +147,8 @@ def _compute_distance_matrix(self):
         """
         if not self.is_connected():
             raise CouplingError("coupling graph not connected")
-        lengths = nx.all_pairs_shortest_path_length(self.graph.to_undirected(as_view=True))
-        lengths = dict(lengths)
-        size = len(lengths)
-        cmap = np.zeros((size, size))
-        for idx in range(size):
-            cmap[idx, np.fromiter(lengths[idx].keys(), dtype=int)] = np.fromiter(
-                lengths[idx].values(), dtype=int)
-        self._dist_matrix = cmap
+        undirected = self.graph.to_undirected()
+        self._dist_matrix = rx.graph_distance_matrix(undirected)
```

Comment:
I probably should update this to be:

```suggestion
        self._dist_matrix = rx.digraph_distance_matrix(self.graph, as_undirected=True)
```

But it might be worth benchmarking it to see if there is a difference perfromance. I wouldn't expect there to be, but I don't remember why I went with this instead of using the kwarg.